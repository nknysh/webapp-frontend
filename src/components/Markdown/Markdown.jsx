import React from 'react';
import { isEmpty } from 'ramda';

import { renderMarkdown } from 'utils/markdown';

import { propTypes, defaultProps } from './Markdown.props';
import { StyledMarkdown } from './Markdown.styles';

export const Markdown = ({ children }) => {
  if (isEmpty(children)) return null;

  const renderedMarkdown = renderMarkdown(children);
  return <StyledMarkdown dangerouslySetInnerHTML={{ __html: renderedMarkdown }} />;
};

Markdown.propTypes = propTypes;
Markdown.defaultProps = defaultProps;

export default Markdown;
