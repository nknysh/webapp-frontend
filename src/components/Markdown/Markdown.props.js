import PropTypes from 'prop-types';

export const propTypes = {
  children: PropTypes.string,
};

export const defaultProps = {
  children: '',
};
