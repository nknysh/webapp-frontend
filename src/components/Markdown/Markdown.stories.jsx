import React from 'react';

import { storiesOf } from '@storybook/react';
import { withKnobs, text } from '@storybook/addon-knobs';

import Markdown from './Markdown';

storiesOf('Markdown', module)
  .addDecorator(withKnobs)
  .add('default', () => {
    return <Markdown>{text('Markdown', '### Some markdown')}</Markdown>;
  });
