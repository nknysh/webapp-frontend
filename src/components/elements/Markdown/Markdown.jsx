import React from 'react';
import { isEmpty } from 'ramda';

import { renderMarkdown } from 'utils';

import { propTypes, defaultProps } from './Markdown.props';
import { StyledMarkdown } from './Markdown.styles';

export const Markdown = ({ children, className }) => {
  if (isEmpty(children)) return null;

  const renderedMarkdown = renderMarkdown(children);
  return <StyledMarkdown className={className} dangerouslySetInnerHTML={{ __html: renderedMarkdown }} />;
};

Markdown.propTypes = propTypes;
Markdown.defaultProps = defaultProps;

export default Markdown;
