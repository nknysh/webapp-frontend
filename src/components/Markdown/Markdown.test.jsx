import React from 'react';

import Markdown from './Markdown';

const getComponent = props => shallow(<Markdown {...props} />);

describe('<Markdown />', () => {
  describe('render', () => {
    it('matches snapshots', () => {
      expect(getComponent()).toMatchSnapshot();
      expect(getComponent({ children: '' })).toMatchSnapshot();
      expect(getComponent({ children: '### test content' })).toMatchSnapshot();
      expect(getComponent({ children: '<p>test content</p>' })).toMatchSnapshot();
    });
  });
});
