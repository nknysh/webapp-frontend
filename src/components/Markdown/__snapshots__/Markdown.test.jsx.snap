// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Markdown /> render matches snapshots 1`] = `""`;

exports[`<Markdown /> render matches snapshots 2`] = `""`;

exports[`<Markdown /> render matches snapshots 3`] = `
<styled.div
  dangerouslySetInnerHTML={
    Object {
      "__html": "<h3>test content</h3>
",
    }
  }
/>
`;

exports[`<Markdown /> render matches snapshots 4`] = `
<styled.div
  dangerouslySetInnerHTML={
    Object {
      "__html": "<p>test content</p>",
    }
  }
/>
`;
